src/provisioner/provisioner.py

```diff
@@ -274,6 +314,27 @@ def cluster_bootstrap(cortx_conf_url: str, force_override: bool = False):
             cortx_conf, node_id, apply_phase, ProvisionerStatus.SUCCESS.value)
         Log.info(f"Finished cluster bootstrap on {node_id}:{node_name}")
 
+    @staticmethod
+    def prepare_delta(_conf_idx: str, _tmp_idx: str):
+        new_keys, deleted_keys, updated_keys = Conf.compare(_conf_idx, _tmp_idx)
+        Conf.load(const.DELTA_IDX, const.DELTA_PATH)
+        for key in new_keys:
+            Conf.set(const.DELTA_IDX, f'new>{key}', Conf.get(_tmp_idx, key))
+        for key in deleted_keys:
+            Conf.set(const.DELTA_IDX, f'deleted>{key}', Conf.get(_conf_idx, key))
+        for key in updated_keys:
+            value = f"{Conf.get(_conf_idx, key)}|{Conf.get(_tmp_idx, key)}"
+            Conf.set(const.DELTA_IDX, f'updated>{key}', value)
+        Conf.save(const.DELTA_IDX)
+
+    @staticmethod
+    def update_conf(cortx_conf: MappedConf, _tmp_idx: str, keys: list):
```

Comment:
If this is internally called function that it must start with _
Can we make function / variable names more meaningful and add description.

Reply:
done